File: 프로그래머스/2/42860. 조이스틱/조이스틱.py
def solution(name):
    answer = 0
    
    num_list = [min(abs(ord('A')-ord(n)), 26-abs(ord('A')-ord(n))) for n in name]
    #상하로 갯수 미리 세서 걍 다 더해놓기.
    answer += sum(num_list)
    min_move = len(name) - 1
    
    for i, c in enumerate(name):
        
        next_i = i+1
        
        while next_i < len(name) and name[next_i] == 'A':
            next_i += 1
        # 각 문자부터 'A..'문자가 있을경우 몇번씩 조이스틱쓰는지 체크
        min_move = min(min_move, 2*i+ len(name)-next_i, 2*(len(name)-next_i)+i)
    return answer+min_move

File: 프로그래머스/2/42860. 조이스틱/조이스틱.js
function solution(name) {
    let answer = 0;
    const min = 65; //A의 아스키코드
    const max = 90 + 1; //Z의 아스키코드 + A->Z로 가는 횟수 1회
    
    const findChar = (find) => {
        //A에서 특정 문자로 가는 최소 거리
        const findCode = find.charCodeAt(); //찾고자하는 문자의 아스키코드
        return Math.min(findCode - min, max - findCode);
    }
    
    const chars = name.split('').map((c) => findChar(c));
    let minMove = chars.length - 1;
    
    chars.forEach((char, idx) => {
        answer += char;
        
        let cursor = idx + 1;
        while(cursor < chars.length && chars[cursor] === 0) cursor++;
        
        minMove = Math.min(minMove, //정 방향으로 전진
                          (idx * 2) + chars.length - cursor, //뒤로 돌아가기
                          idx + 2 * (chars.length - cursor)); //뒤의 요소를 먼저 입력하기
    })
    
    return answer + minMove;
}